test/e2e/config/azure-dev.yaml

```diff
@@ -56,6 +56,8 @@ providers:
         targetName: "cluster-template-machine-pool.yaml"
       - sourcePath: "${PWD}/templates/test/cluster-template-prow-nvidia-gpu.yaml"
         targetName: "cluster-template-nvidia-gpu.yaml"
+      - sourcePath: "${PWD}/templates/test/cluster-template-prow-ci-version.yaml"
```

Comment:
@randomvariable I am unable to reuse `GenerateCIArtifactsInjectedTemplateForDebian` unfortunately, because Azure requires writing the azure.json file for cloud provider and adding that as a Kustomization overrides KubeadmConfig.Files which contains the ci-artifacts.sh script

Reply:
Maybe if we switch the kustomization to a JSON patch it'll work?

Reply:
Just to clarify, you're sending a kustomized template into the function, and then it's overwriting the files section?

Reply:
The issue is that Azure needs the azure.json file to be written. If I add a patch that has a Files entry, it will overwrite the entire Files section, including the ci-artifacts script. I'm not sure if a JSON patch allows merging different entries in the same array field?

Reply:
I think the JSON patch mode allows you to append to arrays using an explicit `add` op, whereas the strategic patch merge will always replace an array.

Reply:
I'll keep the template for now until CAPI supports json patches, then we can switch over to using `GenerateCIArtifactsInjectedTemplateForDebian`